### ISSUE TITLE
Expose forum, news, and radio links on the navbar

### ISSUE THREAD

**Issue by ahmaxed** [freeCodeCamp/freeCodeCamp#41373]
Currently the platform links such as forum, news, and coderadio are bundled under the menu button; however, they should be displayed along side the menu button when there is enough space on the screen.

related: #41032

**Comment by raisedadead** [comment]
_Issue comment https://github.com/freeCodeCamp/freeCodeCamp/issues/40074#issuecomment-791124765 from @QuincyLarson:_

> I have been thinking a lot about this, and I think I have now come to agree with @robole. For the forum, I think it's OK for the freeCodeCamp to lead to forum.freecodecamp.org since that's a different subdomain (and we have the "Visit the curriculum" link on the left hand side of the navigation.)
> 
> But /news is on the same [www.freecodecamp.org](http://www.freecodecamp.org) subdomain, and in many cases people clicking that logo would probably be wondering "what is freeCodeCamp"? (about 80% of traffic to /news comes from Google). So I do think it would be OK to redirect these people to / instead of /news.

**Comment by raisedadead** [comment]
I tend to agree that the logo should take you to the subdomain you are on when clicked, and sub-paths should take you to the parent. That is in this case `/news` should take you to the landing page.

However, this gets very complicated with the languages we have for i18n. As long as we are fine by managing the complexity, we should be fine. 

For me though the intuitive (standard) way on most sites is to take me to the "Home" for that website when I click the logo. Our definition for the home could change with individual language etc.

**Comment by robole** [comment]
Hi,

While I understand that you are considering other changes, I would treat the change @QuincyLarson mentioned about the logo link as an independent change. Changing the href of a link will not impact anything else you are discussing. If you are using this issue for that change, then I suggest changing the title to reflect that.

If you feel you want to do things together. Then, it may be easier to reserve one issue for discussion about the related UI changes, and then have a separate issue for each of the code changes. This will give visibility to others for what actions are taking place and ensure nothing is left out.

All the best

**Comment by ahmadabdolsaheb** [comment]
I think the main issue is that there is no easy way for the users to know which platform they are.

There are links that take the users from one platform to the other; however, if each platform is easily identifiable, users would not have an issue with navigation.

Regardless of what we do, I think we should keep it consistent across platforms. 

Here are the two solutions I see moving forward.
1)  logos navigate to the local homepage( home page o the currently visited platform). In that case we should alter the logo or activate the related navigation tab. example: https://forum.ghost.org/
2) all logos navigate to /, in that case we would need the current platfrom activated in the nav:
https://www.expedia.com/Hotel-Search?adults=2&d1=2021-04-08&d2=2021-04-09&destination=Istanbul%20%28and%20vicinity%29%2C%20Istanbul%2C%20Turkey&endDate=2021-04-09&latLong=41.007884%2C28.977964&regionId=178267&rooms=1&semdtl=&sort=RECOMMENDED&startDate=2021-04-08&theme=&useRewards=false&userIntent

**Comment by raisedadead** [comment]
I really like the former in the two approaches above. It gives out a clear indication of the identity of the application the user is on.

**Comment by ahmadabdolsaheb** [comment]
The changes might affect the SEO on news, but we can add back "news" nav item to resolve that. Additionally, we could highlight or underline the "news" nav to make the nav distinct. 

The changes should be straight forward on news and learn, but the implementation and redesign on the forum might be more time consuming ..

**Comment by robole** [comment]
The question you need to answer is how many websites is freecodecamp? 

At the moment, it looks like 1 or 2 websites (depending on the entry point). The nav links behave like its 3 websites.

Most websites have the same header and same set of navigation links on every page. It is the industry convention. It is the best scenario for the user if you follow this.

Separate websites look different. So, if freecodecamp is more than 1 website, make it _very_ clear that they are going to a different website. Make each of the websites look obviously different -  different colors, logo..etc. The navigation does not have to resemble each other. 

Also, if there is a link to another website in the nav bar, you could also consider adding an icon that indicates its an external link also, making it extra obvious.

![ex](https://user-images.githubusercontent.com/3703647/110521763-20567780-814b-11eb-9d4f-df8ba348f6bc.png)

The preferred destination of a logo link is always the website homepage:
https://www.nngroup.com/articles/homepage-links/

I suggest you talk with an experienced UX designer to get an objective opinion. It sounds like your thinking is very tied to the implementation.

All the best

**Comment by ahmadabdolsaheb** [comment]
@robole, thanks for the feedback. I asked the question on [ux.stackexchange](https://ux.stackexchange.com/questions/137008/where-should-website-logo-link-to-in-an-organization-with-multiple-platforms). I am curious to see the community's response. 

@raisedadead @QuincyLarson, if we end up changing the logo links to / across platforms, it seems the page should offer more customization for the signed in users to make it more visit worthy.

Otherwise linking to platform would be more usable.

**Comment by ahmadabdolsaheb** [comment]
Additionally, any link that redirects to another platform should open in a new tab.( e.g. clicking on items from the search bar on learn should open in a new tab)

**Comment by ahmadabdolsaheb** [comment]
Here is what I propose moving forward. 
The landing page should be treated as an isolated app navigation wise. 
We could expose a list of the related main platforms and select the platform that the user is currently on.

Here is an example:
/              ==>    news  forum curriculum 
news     ==>   <ins>news</ins>  curriculum 
learn     ==>    news  forum coderadio <ins>curriculum</ins> 
forum   ==>    <ins>forum</ins> curriculum 
coderadio ==> news forum <ins>coderadio</ins> curriculum 

We could use a subtle way to show the active nav item so it is not distracting.
I think altering the lower border should suffice. Here are some examples:

<img width="410" alt="Screen Shot 2021-05-10 at 2 53 15 AM" src="https://user-images.githubusercontent.com/4591597/117591039-13e69f00-b13b-11eb-8c9f-f407a78f00c7.png">
<img width="383" alt="Screen Shot 2021-05-10 at 2 53 32 AM" src="https://user-images.githubusercontent.com/4591597/117591041-177a2600-b13b-11eb-9c71-77ebccf13768.png">

**Comment by moT01** [comment]
Is this something we still want @ahmadabdolsaheb?

**Comment by ahmadabdolsaheb** [comment]
@moT01, yes. Optimally we would need coderadio to be a little more stable before exposing it in the main nav.

**Comment by MayurSMahajan** [comment]
First of all I am a beginner to open source and javascript. Is this still an issue, if it is let me tell what I think the issue is: 
Currently the links News, Forum and Radio links all appear in an dialog window, which appears when you click the Menu Button.
Something like this : 
![image](https://user-images.githubusercontent.com/47064215/193641759-37fcc820-1138-4286-94a6-2939d2860fad.png)

The fix for this issue will mean that all these three links will appear **adjacent** to the menu button and **not** inside the dialog box. Is that correct?

**Comment by moT01** [comment]
Yes, this is still an issue @MayurSMahajan. See the [images above](https://github.com/freeCodeCamp/freeCodeCamp/issues/41373#issuecomment-848569978) for how we might want it to look - the bottom two are the main site. I'm not quite sure how we want it, but something close to that I presume. This will probably be a little tricky - everything still needs to be responsive, so the buttons would have to jump back into the main menu when the window is smaller maybe? We don't really have all the details worked out here so there will likely be a lot of back and fourth on the PR. This issue has been here for a while, so I would say to go for it if you want 👍

**Comment by bbsmooth** [comment]
When the platform links are pulled out of the main menu I would use an unordered list and then wrap the list in a `nav` and give it an accessible name with `aria-label="platform"` (using translation for "platform" of course).  Or if not "platform", whatever we want to refer to these links as.

Currently in the curriculum main menu we use an icon to signify that the links to the other platforms open in a new window. While this is not technically a WCAG A/AA requirement (which means we don't have to do it to be considered accessible) it is a AAA requirement and almost every accessibility expert you ask will highly recommend it. But I'll leave it up to consensus as to whether we want to include these icons when these platform links are pulled out of the main menu.

Since it appears (based on the mock-up images above) that we are also going to include the current platform in those links (for example, there will be a "curriculum" link showing even when the user is somewhere in the curriculum) I'm assuming that that link would not open in a new window?  Also, I think we should add `aria-current="true"` to the current platform link since these links are in a set and we are visually identifying which one is current with a focus indicator below the link.

**Comment by ahmadabdolsaheb** [comment]
Just a little update on the issue. We would need to expose the platform links on desktop screen sizes. Since the space is limited in the navigation bar (different translations might require more space), I would be cautious about adding the icons for the exposed platform nav items. 

Additionally @bbsmooth , I discussed @oriooctopus exposing the language selector; However, it would be reduced to an icon. I was wondering about your thoughts?

**Comment by MayurSMahajan** [comment]
Okay guys! I am working on this issue.

**Comment by oriooctopus** [comment]
Regarding the language selector, I think that if we accompany the icon with the initials of the language, that would be the most clear. Regarding concerns about this taking too much space, the accompanying text could disappear below a certain breakpoint. Perhaps it only shows up above tablets width and below only the icon shows.

I like [this example](https://miro.medium.com/max/1400/1*QnfCBUQrHMMlgHF1nm4PxA.gif), although in our case the initials would not be vertically stacked, they would be next to the globe.

Something like this
![Screen Shot 2022-10-09 at 4 34 36 PM](https://user-images.githubusercontent.com/20331238/194776415-c52dcb5d-7f02-4df7-9cf5-b96386bf289e.png)

**Comment by moT01** [comment]
That language selector and icon look real good @oriooctopus. I feel like I recall a conversation about adding a language selector in the nav and we decided against it because you don't really need to switch languages much. Once you are on the language you want, you don't need to change anymore.

**Comment by oriooctopus** [comment]
@moT01 In general, language switching is not something people do often, but there's a reason it is located in the navbar on almost every website that supports it. The issue is not with easily accessing the language switcher, **it's about knowing that you can switch at all**. I had no idea that free code camp supported other languages for a long time because that functionality was buried in the menu. I'd be willing to bet that many other people who are not native english speaker never got to use the translations of free code camp simply because they didn't know it existed. 

I don't think we should break convention here. MDN, for example, which is very similar to FreeCodeCamp, has a languagae switcher in their menu, even though I imagine it is not used more than once.

As a side note, from looking at the google analytics data on translation usages for FreeCodeCamp, we have a ton of people using FreeCodeCamp from non english countries, yet very small amounts of people using the translated versions of FreeCodeCamp. My take on that is it supports the idea that people don't know that FreeCodeCamp has translations. After all, many coding resources don't have translations so that's a pretty intuitive conclusion.

**Comment by oriooctopus** [comment]
As a side note, from looking at the google analytics data on translation usages for FreeCodeCamp, we have a ton of people using FreeCodeCamp from non english countries, yet very small amounts of people using the translated versions of FreeCodeCamp. My take on that is it supports the idea that people don't know that FreeCodeCamp has translations. After all, many coding resources don't have translations so that's a pretty intuitive conclusion.

**Comment by bbsmooth** [comment]
> Additionally @bbsmooth , I discussed @oriooctopus exposing the language selector; However, it would be reduced to an icon. I was wondering about your thoughts?

There is no WCAG violation for using an icon button without text. I like the idea of adding the two letter language code next to it as that might make it a little more clear, but that's not required. If we think that people aren't realizing that other languages are supported because the language switcher is in the menu then I definitely agree that it's a good idea to move it out as a separate item on the nav bar.

**Comment by ojeytonwilliams** [comment]
One way we could advertise the available languages is by showing a one-time modal giving people the option to switch.  @ahmadabdolsaheb this might be worth A/B testing to check that it doesn't frustrate people who just want to get started.

Regardless of whether or not we do that, I like @oriooctopus's suggestion of separating out the switcher for more visibility. We've plenty of space for a globe and a couple of characters.

**Comment by ahmadabdolsaheb** [comment]
@ojeytonwilliams, possibly after the completion of a few challenges. I wonder if there is a similar convention for that. 

Since all seem to be in agreement, I think we should move forward with the icon and the Initials.
I only wonder if the English initials are meaningful for speakers of other languages?

**Comment by ojeytonwilliams** [comment]
> I only wonder if the English codes are meaningful for speakers of other languages?

Probably not.  At least not for everyone.  I think it makes sense to keep doing it how we're are currently.  The question is how to represent the menu itself.  Just a globe?  I've seen a few icons used, like https://en.wikipedia.org/wiki/Language_Icon, but I've no idea how universal they are.

**Comment by ahmadabdolsaheb** [comment]
Let's move forward with just a globe or a globe and  ISO language codes. Either way works for me.

**Comment by ojeytonwilliams** [comment]
I'm not keen on the ISO codes, honestly.  They're useful for developers, but using Latin characters for something that can be localised feels like a mistake.

**Comment by ahmadabdolsaheb** [comment]
Thanks for the feedback. Then we would want to have the currently used globe icon in the navigation.

**Comment by MayurSMahajan** [comment]
Hey guys! I have made some progress. I created a separate component for the links, and exposed it outside of the menu button. All of them are functional meaning if you click them they go to the desired page. This is how it looks.

![Screenshot (1)](https://user-images.githubusercontent.com/47064215/198257286-7fb5d8a6-7937-47ef-8abc-80170c48b568.png)

I still have to make it responsive. Once that is done then the issue will be solved.
I have made all these changes in GitPod, I have never used it before. I committed my changes but I am unable to sync changes. I think it is because I have been modifying the original freecodecamp workspace and not the fork that I created. Is there some way I can get my progress on my fork.

**Comment by MayurSMahajan** [comment]
After the initial issue is solved I will try exposing the Language selector as well.

**Comment by ahmadabdolsaheb** [comment]
Thanks for working on this.
So the curriculum need to be added to the list =) (so users know where they are)
You could signify it being active with the a button border.
Also, the platform icons could go in the interest of saving space.

**Comment by MayurSMahajan** [comment]
Alright! I have added curriculum and removed the platform icons.

**Comment by oriooctopus** [comment]
Hey there, what's the status of this? I can help out with the language selector side of this if @MayurSMahajan is done

**Comment by MayurSMahajan** [comment]
@oriooctopus I am struggling with making things reponsive, the sign in button is causing a lot of problems. Should I send a pull request with my current progress?

**Comment by oriooctopus** [comment]
@MayurSMahajan @ojeytonwilliams  I'll try to pull down the PR this weekend and take a look at things

**Comment by MayurSMahajan** [comment]
sure

**Comment by laportabellita** [comment]
Overview 

Hello, my name is Sofia, I’m @oriooctopus ’s wife. He and @ahmadabdolsaheb had a meeting about a second round of changes to the design of the nav bar. What was talked about was making the donations button the primary button, formalizing the design for the language picker, changing the sign in text button to an icon button, and changing the “Menu” button to show a hamburger icon instead of the word. 

After taking into account the points discussed and making my own observations, I've created a new design with the following changes:

 
- Just like in the first version of the new design I've put the “News”, “Forum”, “Curriculum”, and “Radio” outside of the menu for visibility
- Also, in line with what was discussed earlier this issue I added the icon for the language switch to the navbar, so that non-english speakers can easily find  it and use it to switch to their preferred language
- For the user login, I also used an icon, because it’s universally recognizable across languages and it’s a common pattern across different sites. In general, using icons instead of words saves up space, which is very useful since there’s several items that we’re trying to include in the nav bar. Another benefit is to increase consistency of experience because the existing implementation currently on the website already shows the profile icon when the user is logged in, so this is in line with that pattern . 
An example of a popular website using this pattern is [Airbnb](https://www.airbnb.com/):  

<img width="624" alt="Screen Shot 2022-12-08 at 11 52 46 AM" src="https://user-images.githubusercontent.com/82290980/206542269-5fe686ab-7ebc-4fe8-9624-4e68f3629efe.png">

- I incorporated the “Donate” button as a primary button, placing it at the right side of the nav bar in order to make it a prioritized action. This will be helpful when it comes to encouraging donations.  

- I also made some visual tweaks, like increasing the height by a bit, which makes it look less crowded, and more visually pleasing.

The following is the first iteration of designs I came up with: 
<img width="505" alt="Screen Shot 2022-12-08 at 11 54 53 AM" src="https://user-images.githubusercontent.com/82290980/206542676-41ed7b6e-192c-46fa-b0fe-b8f422a7a9b1.png">

After the initial round of design I realized we couldn’t make space for everything. Although the above screenshot looks OK, there are a number of situations that would make it not work. The same content in other languages like german would crowd things even more. Another important aspect to consider is that the menu would be inflexible, because if more content was ever created and a necessity to add it to the menu arose, it would’ve been impossible to include that content. Most importantly, on smaller screen sizes there would have definitely not been enough space. @[ahmadabdolsaheb](https://github.com/ahmadabdolsaheb)’s original design looks less crowded because it was on a very wide screen. The screenshot above however is 1440px wide, which is actually on the larger side and it has to be able to go as small as 1100px. 

Keeping this in mind, I applied some changes to our initial design: 

1. I shrunk the search bar, extending it whenever a user clicked on it to show the full version. If you want to see that in action I created an interactive prototype on figma, which is linked below. The below screenshot only shows it in its non-expanded state.
2. I moved the menu items to the left side of the page to give them more room. 
3. I added a hamburger menu, which displays three elements: 
 - Night Mode
- Settings
- Log out (for logged in users) 


- Desktop after the mentioned changes: 

<img width="625" alt="Screen Shot 2022-12-08 at 11 55 53 AM" src="https://user-images.githubusercontent.com/82290980/206542917-866921c7-2ea1-4ee1-857d-8203a805b0fa.png">



- Tablet after changes: 

<img width="623" alt="Screen Shot 2022-12-08 at 11 56 07 AM" src="https://user-images.githubusercontent.com/82290980/206542977-68ca1d35-1c66-4cee-bb84-c0c1baa3f0ce.png">



By the way, in the current design the search bar is not included in the navbar for the tablet view, but I believe we have enough space to include it if that’s something you would like. 
[This site](https://docusaurus.io/) has it and it served as an inspiration for this second re-design. Take a look at it on tablet/mobile. I think it would be good to do the same for our site.

- Phone after changes:

 Iphone 8 (1) : 
<img width="400" alt="Screen Shot 2022-12-08 at 11 56 29 AM" src="https://user-images.githubusercontent.com/82290980/206543077-f31530fb-746f-4541-9559-05b7178560da.png">


Iphone 8 (2) Closed and opened menu:
<img width="398" alt="Screen Shot 2022-12-08 at 11 56 53 AM" src="https://user-images.githubusercontent.com/82290980/206543154-3a2bfb49-c472-4676-9a2f-173c71986070.png">


Iphone SE (closed and open menu) :
<img width="304" alt="Screen Shot 2022-12-08 at 11 57 21 AM" src="https://user-images.githubusercontent.com/82290980/206543255-5b62d4ad-f344-46c3-a633-a5f9f6e3ac23.png">



[Here’s a link to the figma file](https://www.figma.com/file/5CzMtYAXczx4unvriYUtVR/FreeCodeCamp-NavBar-(Copy)?node-id=101%3A64&t=nKaRaMANhRLsyS1r-1). By the way, you can view the different sizes in prototype mode to interact with the search bar, and the icon for the night mode (it switches from a sun to a moon on click)

**Comment by MayurSMahajan** [comment]
Okay, First of all, excellent design there. I love the suggested changes! I will include all of them in my PR, and hopefully, I'll inform you soon.

**Comment by ahmadabdolsaheb** [comment]
@laportabellita , thank you  for creating  Figma mocks and for documenting your design process. Here are my thoughts:

- It seems we would need to introduce dynamic breakpoints where the navigation UI changes based on the width of the localized nav items. 
- I think moving the platform buttons to left is a good idea, but we would need the dynamic breakpoints in place first and the current platform (curriculum) needs to be selected with an underline preferably.
- Replacing the sign in call to action with an icon is a bit tricky as it would have a direct effect on our sign ups. We should deploy it in stages and monitor the changes. There are a few untested hypotheses in regards to the donation section of the navigation. Perhaps donation should be a button on the navigation and only turned into a call to action when the user is signed in and has not donated.
------
Here are the action items as of now:
- [ ] Research dynamic breakpoints (it might have another name) best practices for localized navigation.
- [ ] Expose the Donate button for large and medium sized screens. It should say donate regardless of the donation state. Additionally, the button should always be "Donate" on small screen dropdown for consistency. 
- [ ] The menu button should be replaced with an icon as it seems to be the convention.
- [ ] Language selector needs to be exposed in the nav as an icon. After @bbsmooth's confirmation, we could wire the button to open a modal instead of a dropdown similar to Airbnb. 
- [ ] The height of the navigation should increase to 60px. That will also give enough spacing for tabs in the mobile challenge layout. That being said, the change would require rigorous testing to ensure it is not affecting other parts of the platform.
------
Thank you for patience @MayurSMahajan. It would be great if your first pr addresses exposing the donate button as it would have the largest impact. Other prs should address only one feature at a time to make reverting and altering them more straightforward down the line.

**Comment by laportabellita** [comment]
@[ahmadabdolsaheb](https://github.com/ahmadabdolsaheb) thanks for checking my considerations for the navbar! Here are some notes and comments: 

1. When you say "Here are the action items as of now", these comments are for @MayurSMahajan, not for me, correct? 

**2. It seems we would need to introduce dynamic breakpoints where the navigation UI changes based on the width of the localized nav items**
**3. I think moving the platform buttons to left is a good idea, but we would need the dynamic breakpoints in place first and the current platform (curriculum) needs to be selected with an underline preferably**

Can you clarify these two points?  I thought the current navbar had dynamic breakpoints already

**4. Replacing the sign in call to action with an icon is a bit tricky as it would have a direct effect on our sign ups. We should deploy it in stages and monitor the changes. There are a few untested hypotheses in regards to the donation section of the navigation. Perhaps donation should be a button on the navigation and only turned into a call to action when the user is signed in and has not donated:** 

This makes sense, my only concern would be that It would be weird to have two primary buttons (If our plan is to make donate be a primary button), perhaps we could modify the sign in button to be a secondary button? 

**5. We could keep the platform links, search in the dropdown menu for the small screens**

This makes sense too, do you want me to make a design for this?

**Comment by ahmadabdolsaheb** [comment]
> When you say "Here are the action items as of now", these comments are for @MayurSMahajan, not for me, correct?

Those are for anyone who is interested in contributing. Feel free to dive into the first action item or any of the other ones if you interested.

> Can you clarify these two points? I thought the current navbar had dynamic breakpoints already.

So we have a few nav items that have different widths when translated to different languages. 
Rather than guessing where to put the breakpoints, we would need to have a system in place that renders the appropriate layout to users when necessary.  

For instance, Chinese platform nav items are narrower than German platform nav items. That would allow us to keep showing them in the nav bar for tablet screens while the German items need to be folded in the dropdown menu in the same width. 

We would need to know what the best practices are when the width of the nav items are not constant. Once that is taken care of we can proceed to exposing the platform nav items.

> This makes sense, my only concern would be that It would be weird to have two primary buttons (If our plan is to make donate be a primary button), perhaps we could modify the sign in button to be a secondary button?

You are right. We need to have one primary call to action at a time. As a result,  we are going to expose the donate button first( not call to action ). And then,  a/b test the rest of the changes to see how that would effect sign ups.

> This makes sense too, do you want me to make a design for this?

It is not necessary at this point. Once the current action items are taken care of, we would need to make slight modifications to the mocks (item widths, choice of icons, search bar animation, etc.) to represent the final stage. 

On the side note, what do you think about having the menu and profile image be the same button and gradually drop the menu icon all together?

**Comment by oriooctopus** [comment]
> Replacing the sign in call to action with an icon is a bit tricky as it would have a direct effect on our sign ups. We should deploy it in stages and monitor the changes. There are a few untested hypotheses in regards to the donation section of the navigation. Perhaps donation should be a button on the navigation and only turned into a call to action when the user is signed in and has not donated.

I definitely agree about deployment in stages and monitoring the analytics. I personally think that the donation button should always be exposed as a primary button but that's the beauty of having all these analytics; we can see what the effect between the two options would be

**Comment by laportabellita** [comment]
1. Replacing the sign in call to action with an icon is a bit tricky as it would have a direct effect on our sign ups. We should deploy it in stages and monitor the changes. There are a few untested hypotheses in regards to the donation section of the navigation. Perhaps donation should be a button on the navigation and only turned into a call to action when the user is signed in and has not donated.
As for the donate button: Do you mean the primary button would change from “Sign in” to “Donate” after they sign in?, and Sign in would disappear altogether?

2. So we have a few nav items that have different widths when translated to different languages.
Rather than guessing where to put the breakpoints, we would need to have a system in place that renders the appropriate layout to users when necessary.
For instance, Chinese platform nav items are narrower than German platform nav items. That would allow us to keep showing them in the nav bar for tablet screens while the German items need to be folded in the dropdown menu in the same width.

This would probably be very responsive and a good idea. Unfortunately, I didn’t find much information about navbars changing for different languages specifically. From a design perspective it’s not hard, but the implementation could possibly get a little complicated, what do you think?[ https://github.com/MayurSMahajan](https://github.com/MayurSMahajan) 
My opinion is that the navbar should just be made flexible enough to handle different languages but there shouldn't be any different code for the different languages. 

3. About this: “You are right. We need to have one primary call to action at a time. As a result, we are going to expose the donate button first( not call to action ). And then, a/b test the rest of the changes to see how that would affect sign ups.” 

Are you saying that we should keep the donate button, but not as a primary button? If so, do you want me to tweak my design to do so? 

4. On the side note, what do you think about having the menu and profile image be the same button and gradually drop the 
menu icon all together?

<img width="395" alt="Screen Shot 2022-12-14 at 9 27 30 AM" src="https://user-images.githubusercontent.com/82290980/207652179-24d02a02-daac-4011-81e5-94f3879c89aa.png">
Do you mean something like this? (This is Airbnb). I think this could be a good idea, in fact I was trying something like this in a previous design iteration, but I was concerned that it wouldn’t match with the style of the rest of the nav. I can try out a couple more designs and get back to you with something later.

**Comment by ahmadabdolsaheb** [comment]
> As for the donate button: Do you mean the primary button would change from “Sign in” to “Donate” after they sign in?, and Sign in would disappear altogether?

Yes. I am thinking out loud here but there other option could be replacing then the sign in button with the avatar when the user signs in. At the same time, the navigation Donate button would turn into CTA.

> This would probably be very responsive and a good idea. Unfortunately, I didn’t find much information about navbars changing for different languages specifically. From a design perspective it’s not hard, but the implementation could possibly get a little complicated, what do you think?[ https://github.com/MayurSMahajan](https://github.com/MayurSMahajan)
My opinion is that the navbar should just be made flexible enough to handle different languages but there shouldn't be any different code for the different languages.

At this point I am just wondering what the best practices are. The code would be the same, but the breakpoints would be based on the width of the content. 

> Are you saying that we should keep the donate button, but not as a primary button? If so, do you want me to tweak my design to do so?

Yes, feel free to add that to the design. 

> Do you mean something like this? (This is Airbnb). I think this could be a good idea, in fact I was trying something like this in a previous design iteration, but I was concerned that it wouldn’t match with the style of the rest of the nav. I can try out a couple more designs and get back to you with something later.

Yes, we could look into different iterations to see if there is one that fits us better.

**Comment by MayurSMahajan** [comment]
@laportabellita I too think it is a tad bit complicated to implement different breakpoints with varying sized nav items. I agree that we should not write code for each language and instead work on a system that adapts to the nav items size. I am currently researching about how we could achieve a system that introduces the required dynamic breakpoints.

**Comment by oriooctopus** [comment]
@MayurSMahajan I would just use media queries for the breakpoints.

**Comment by laportabellita** [comment]
I created a few initial designs for each of the iterations we were considering: You can look at the [figma file here](https://www.figma.com/file/5CzMtYAXczx4unvriYUtVR/FreeCodeCamp-NavBar-(Copy)?node-id=197%3A254&t=Qdz1fPkZwkJCqadz-1)

First, I made the navbar that users would see when they’re not logged in. As discussed, the primary button would be the Sign in one, and the Donate button would be the secondary one. 
After they log in, the primary button changes to show the word “Donate”, and the Sign in button disappears altogether, getting replaced for the user icon.
I’m not sure this is a great idea, because I believe it could potentially confuse the users, since they might not notice the change, making it less likely for them to donate. If you think we should continue in this direction, maybe we could take some actions to prevent the user from missing the change, like moving the button. Still, I believe it is best to keep the Donate button consistent, and just decide if we want to make it a priority by making it primary or not. 

I made two versions of the state of the nav while the user is not logged in, one where the log in and the menu icons share a container, and one where they’re separated from each other. 

For the logged in state of the navbar, I made several different designs, playing with the secondary button. Between the last two the only difference is a drop shadow that’s hard to see on the screenshots.   

These are all initial iterations, once we agree on the functionality, I can create refined versions of our final choice. 

<img width="619" alt="Screen Shot 2022-12-15 at 2 49 47 PM" src="https://user-images.githubusercontent.com/82290980/207974586-4d5f2208-7e04-4608-8054-60b639fb4d74.png">



Additionally, I created three versions of the container with the login and the menu icons, which I would also refine if we decide to apply this idea.

<img width="1058" alt="Screen Shot 2022-12-15 at 2 50 47 PM" src="https://user-images.githubusercontent.com/82290980/207974711-e5c16f19-61b2-49aa-a0c0-ca0960d4dad2.png">

**Comment by Sboonny** [comment]
> These are all initial iterations, once we agree on the functionality, I can create refined versions of our final choice.
> 
> <img alt="Screen Shot 2022-12-15 at 2 49 47 PM" width="619" src="https://user-images.githubusercontent.com/82290980/207974586-4d5f2208-7e04-4608-8054-60b639fb4d74.png">
> 
> Additionally, I created three versions of the container with the login and the menu icons, which I would also refine if we decide to apply this idea.
> 
> <img alt="Screen Shot 2022-12-15 at 2 50 47 PM" width="1058" src="https://user-images.githubusercontent.com/82290980/207974711-e5c16f19-61b2-49aa-a0c0-ca0960d4dad2.png">

What about instead of using donate button, we use an icon with an empty heart instead, and if camper donate then the heart fills up.

Here is an empty heart icon: https://fontawesome.com/icons/heart?s=regular&f=classic

---

if we apply the empty heart icon, we can add hidden eastern eggs, here are examples:

- if they use CamperMode, we fill the icon with music icon.
- if they earn the Resposnive web design certification, we fill the icon with certification icon.
- if they earn every certification, we fill it will freeCodeCamp logo in it.

**Comment by ahmadabdolsaheb** [comment]
@MayurSMahajan Thank you for doing the researching. I am open to using just breakpoints as @oriooctopus mentioned for its simplicity, but we would need to know the possible solutions before committing to one. Also, since exposing the donate button action item is time sensitive, I was wondering if you are interested in altering your pr to the spec mentioned above. If so, let me know as we would need to land this feature next week.

@laportabellita, thank you for visualizing our options. I agree that switching the buttons might cause confusion. Ultimately replacing the sign in button with an avatar is more desirable, and we will gradually make the changes in that direction and observe the user behavior. 

@Sboonny, I would love to replace the donate button with an icon, but sadly no icon is widely associated with making donations :/

**Comment by oriooctopus** [comment]
> I was wondering if you are interested in altering your pr to the spec mentioned above. If so, let me know as we would need to land this feature next week.

Is this directed at @MayurSMahajan?

**Comment by MayurSMahajan** [comment]
Okay I will expose the donate BTN and update my PR in 1-2 days.

**Comment by ahmadabdolsaheb** [comment]
> Is this directed at @MayurSMahajan?

yes

**Comment by oriooctopus** [comment]
Have there been any updates on this issue? I saw that @ahmadabdolsaheb merged something but I didn't notice any changes to the navbar

**Comment by MayurSMahajan** [comment]
Ahmad's PR has exposed the Donate button. I am trying to work on the rest, but I cannot due to other time commitments. If someone wants to work on this issue, they can.

**Comment by ahmadabdolsaheb** [comment]
@oriooctopus, we reversed that since we needed to add some change to the donor state.

**Comment by oriooctopus** [comment]
@ahmadabdolsaheb I guess I can take this on

**Comment by oriooctopus** [comment]
@ahmadabdolsaheb  So I've started working on this. What I'm essentially adding is an IconButton and a Dropdown. In accordance with https://github.com/freeCodeCamp/freeCodeCamp/issues/44668 I want to lean on the component library as much as possible, and even create new components if necessary. 

Regarding the dropdown: There is existing behavior on the navbar set by the `nav-list` class that can be used and refactored to be shared between the two dropdowns (and future dropdowns). I also noticed https://github.com/freeCodeCamp/freeCodeCamp/pull/47043 but see it hasn't merged yet. Do you know when it will be merged? That could potentially be a blocker for this work

Regarding the IconButton: I plan on creating a component for this.

**Comment by ahmadabdolsaheb** [comment]
@oriooctopus, I should be able to get to the outstanding component library prs this week.

**Comment by Sboonny** [comment]
There were no push back against adding the logo as glyph in this issue #49985, if exposing the buttons will cause us to need more spacing, changing it to glyph in the future is a valid option.

**Comment by ahmadabdolsaheb** [comment]
@Sboonny this should do: https://commons.wikimedia.org/wiki/File:Globe_icon.svg

**Comment by ahmaxed** [comment]
Since we have achieved to expose the localization dropdown, I propose we move forward with the current design for simplicity.

I will add an AB test for the donation button down the line.
